### PR TITLE
Make computed.or and computed.and support property expansion

### PR DESCRIPTION
**Pull request by ming-codes**
`computed.or` and `computed.and` already supports more than two properties. I added some test cases for this use case.

Added code to pass `getProperties` through `Ember.expandProperties` to further enhance the above case.

### CODE REVIEW

File: packages/ember-metal/lib/computed_macros.js [hunk 1]
```diff
@@ -14,7 +15,9 @@ import alias from 'ember-metal/alias';
 function getProperties(self, propertyNames) {
   var ret = {};
   for (var i = 0; i < propertyNames.length; i++) {
-    ret[propertyNames[i]] = get(self, propertyNames[i]);
+    expandProperties(propertyNames[i], function (name) {
```
Comment (rwjblue) [inline packages/ember-metal/lib/computed_macros.js]: This is triggering a JSHint failure for "don't make functions in a loop"

**Comment by ming-codes** [comment]
@rwjblue All tests passed

**Comment by rwjblue** [comment]
Looks good, can you update the docs for `computed.or` and `computed.and` to add an example using brace expansion and also squash commits?

**Comment by rwjblue** [comment]
Hmm, actually, I just had a thought.  It seems like we should call `expandProperties` upon creation of the macro (inside `generateComputedWithProperties` [around here](https://github.com/ming-codes/ember.js/blob/computed-property-expansion/packages/ember-metal/lib/computed_macros.js#L31)).  The current implementation seems like it would be calling `expandProperties` on every dependent key invalidation (which seems like a bit of wasted work).

**Comment by ming-codes** [comment]
`computed.and` doesn't seem to follow the JavaScript-like behavior of `computed.or`. As in, `computed.and` always coerce to `false` rather than the original DK's falsy value. I wonder if this is a bug so I can fix it.

Test case below

```js
  var obj = { one: true, two: true };
  defineProperty(obj, 'oneAndTwo', and('one', 'two'));

  set(obj, 'one', null);
  set(obj, 'two', 'Yes');

  equal(get(obj, 'oneAndTwo'), null, 'returns falsy value as in &&'); // expects `null` as JavaScript expression `null && 'Yes'` eval to `null`
```

**Comment by rwjblue** [comment]
Yes, I believe that it should behave the same as `computed.or` (and normal JS)

**Comment by ming-codes** [comment]
@rwjblue Done. I think github let you squash and merge now. Or I can do it. Let me know.

**Comment by rwjblue** [comment]
@ming-codes thank you!

**Pull request merged**